### PR TITLE
adding seissol_output_extractor to seissolxdmfwriter

### PR DESCRIPTION
**Pull request by Thomas-Ulrich**
(sorry for another PR of seissolxdmfwriter).
I thought that I could take advantage of the latest change in seissolxdmfwriter to rewrite in few lines this script:
https://github.com/SeisSol/SeisSol/blob/master/postprocessing/visualization/tools/extractDataFromUnstructuredOutput.py

But I realized in the process that  I would need to modify seissolxdmfwriter because  extractDataFromUnstructuredOutput.py reads and writes on the fly (to avoid loading 2Tb of memory all at one).

Anyway, I ended up doing that and fixed a few bugs in the process.

The new extractor (seissol_output_extractor) will be shipped with the module, which is quite convenient.
Compared to extractDataFromUnstructuredOutput, it supports writing partition, fault-tag, locationFlag (and write them only once).
It enables compression by default (because of that is is a bit slower than extractDataFromUnstructuredOutput.py).
Without compression, it is still slightly slower (nothing very significant).
It has a nice tqdm interface (maybe less nice when writting to a log).

### CODE REVIEW

File: seissolxdmfwriter/README.md [hunk 1]
```diff
@@ -30,5 +30,25 @@ sxw.write(
...
+    compression_level=4,
+)
+
 ```
 
+The module also encapsulates `seissol_output_extractor`, which can be used to extract and process data from SeisSol output files, allowing selection of variables, time steps, spatial ranges, and output format.
+Here is an example of use:
+
+```bash
+seissol_output_extractor test-fault.xdmf --time "i2,i4" --variable PSR Vr partition
```
Comment (sebwolf-de) [inline seissolxdmfwriter/README.md]: Can you write a bit more about the features? E.g. how do  I extract spatial ranges, what's the `i2,i4` stand for ...

File: seissolxdmfwriter/seissolxdmfwriter/__init__.py [hunk 2]
```diff
@@ -1,2 +1,2 @@
 from .seissolxdmfwriter import *
-__version__ = '0.2.2'
+__version__ = '0.2.3'
```
Comment (sebwolf-de) [inline seissolxdmfwriter/seissolxdmfwriter/__init__.py]: It is a new feature, so IMHO, we could go for 0.3.0.

File: seissolxdmfwriter/seissolxdmfwriter/seissol_output_extractor.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    "--precision",
+    type=str,
+    choices=["float", "double"],
+    default="float",
+    help="precision of the data in the output file",
+)
+parser.add_argument(
+    "--backend",
+    type=str,
+    choices=["hdf5", "raw"],
```
Comment (sebwolf-de) [inline seissolxdmfwriter/seissolxdmfwriter/seissol_output_extractor.py:51]: raw -> posix. At least that's the termionology in the SeisSol parameter file.

**Review by sebwolf-de — APPROVED** [review]
LGTM, only a few comments

File: seissolxdmfwriter/seissolxdmfwriter/seissol_output_extractor.py (hunk 3 above)
Comment (Thomas-Ulrich) [reply seissolxdmfwriter/seissolxdmfwriter/seissol_output_extractor.py:51]: See comment of Lukas:
https://github.com/SeisSol/SeisSol/pull/297#discussion_r580114139

**Pull request merged**